### ISSUE TITLE
regionmask.mask concatenate issue

### ISSUE THREAD

**Issue by sarahcfn** [mathause/regionmask#36]
Hi,
I have a shapefile of polygons and multipolygons [Peatlands.zip](https://github.com/mathause/regionmask/files/3561344/Peatlands.zip) from [here](http://archive.researchdata.leeds.ac.uk/251/ ). 
When applying the Regions_cls function, I get a value error 
[Region_mask_Value_error.txt](https://github.com/mathause/regionmask/files/3561331/Region_mask_Value_error.txt)

regionmask.__version__
Out: '0.4.0'
I tried to explode the DF before applying the mask, but without success. Here is my code:
`O_USA_peat_path = '.../Other_North_American_Peatlands.shp'

DS = gpd.read_file (O_USA_peat_path)

DS.crs = {'init' :'epsg:4087'}

DS=DS.to_crs({'init': 'epsg:4326'})

region_names = ['Peatland_{}'.format (i) for i in range (len(DS))]

g_region_sub = [i for i in DS.geometry]

Abbreviations_P = ['Peatland_{}'.format (i) for i in range (len(DS))]

region_numbers = np.arange(len(region_names))

Regionmask= regionmask.Regions_cls('All',region_numbers,region_names,Abbreviations_P,g_region_sub)

lon =  np.arange(-179.7, 180.2,.5)

lat = np.arange(-89.6, -59.9,.5)

Grid_mask = Regionmask.mask(lon, lat)`

Because the Regionmask coords method is not working, I can't generate the mask in the last step. 
Any help would be very much appreciated! 

Many thanks!

**Comment by mathause** [comment]
Thanks for adding the dataset. Peatland includes a z-coordinate (seems to always be 0) which regionmask cannot handle. You can get rid of this coordinate like so:

``` python
import shapely 

def _to_2d(x, y, z):
    return tuple(filter(None, [x, y]))

def to2d(shape):
    return shapely.ops.transform(_to_2d, shape)

g_region_sub = [to2d(i) for i in DS.geometry]
```
However, peatland has almost 40'000 regions so `regionsmask` takes *very long. #24 may help.

Also you will not get many reagions that actually match. Further the peatland regions seem to be very small, so even with a grid spacing of 0.5 ° you will get few matches.

**Comment by sarahcfn** [comment]
Thanks so much, the #24 worked very well! 

I am aware of the issue with the small polygons, unfortunately I can't consider them (though the number of matches is still acceptable for my application).

**Comment by mathause** [comment]
#24 is implemented in v0.5.0, so this should work in regionmask
TODO: add a test for a dataset with a z-coordinate
